### PR TITLE
test: Reduce noise in test logging

### PR DESCRIPTION
**Pull request by david-allison**
## Purpose / Description
Lots of noise in test logs

## Fixes
Comment: https://github.com/ankidroid/Anki-Android/pull/13345#issuecomment-1445219437

## Approach
* Avoid 'Op on UI Thread'
   * Stop logging this one under Robolectric - embedded in `Anki-Android-Backend`
* Avoid 'Don't know what to do with dataSource..'
  * Answer from GitHub `https://github.com/robolectric/robolectric/issues/4673`

## How Has This Been Tested?
Test only

## Learning (optional, can help others)
Note: One gotcha with this shadow is that you need to either configure an exception or a [ShadowMediaPlayer.MediaInfo](https://robolectric.org/javadoc/3.8/org/robolectric/shadows/ShadowMediaPlayer.MediaInfo.html) instance for that data source (using [addException(DataSource, IOException)](https://robolectric.org/javadoc/3.8/org/robolectric/shadows/ShadowMediaPlayer.html#addException-org.robolectric.shadows.util.DataSource-java.io.IOException-) or [addMediaInfo(DataSource, MediaInfo)](https://robolectric.org/javadoc/3.8/org/robolectric/shadows/ShadowMediaPlayer.html#addMediaInfo-org.robolectric.shadows.util.DataSource-org.robolectric.shadows.ShadowMediaPlayer.MediaInfo-) respectively) before calling [setDataSource(org.robolectric.shadows.util.DataSource)](https://robolectric.org/javadoc/3.8/org/robolectric/shadows/ShadowMediaPlayer.html#setDataSource-org.robolectric.shadows.util.DataSource-), otherwise you’ll get an IllegalArgumentException.
> https://robolectric.org/javadoc/3.8/org/robolectric/shadows/ShadowMediaPlayer.html#addMediaInfo-org.robolectric.shadows.util.DataSource-org.robolectric.shadows.ShadowMediaPlayer.MediaInfo-

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidApp.kt [hunk 1]
```diff
@@ -105,7 +105,7 @@ open class AnkiDroidApp : Application() {
         CrashReportService.initialize(this)
         if (BuildConfig.DEBUG) {
             // Enable verbose error logging and do method tracing to put the Class name as log tag
-            Timber.plant(DebugTree())
+            Timber.plant(TimberDebugTree.createInstance())
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidApp.kt:108]: Hmm - the point of that one is that eventually there should be zero of those, so they are potentially useful in DEBUG in the way a KotlinCleanup could be useful, but in the automated test context they are less useful at the moment. Is it possible to only filter those in a robolectric context but leave on for the general debug case?

**Review by mikehardy — COMMENTED** [review]
I'm iffy on the UI thread op stuff, I feel like those are potential ANR cases or at least will lead to UI hangs, so they could be good to leave on - and those at least are a single `grep -v` to get rid of at the moment...

File: AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidApp.kt (hunk 1 above)
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidApp.kt:108]: That's exactly what `createInstance` does

**Review by mikehardy — APPROVED** [review]
Okay, not it is blindingly obvious 😆

**Pull request merged**